pkg/controller/backupinfrastructure/backup_infrastructure_control.go

```diff
@@ -0,0 +1,372 @@
+
+	// We create botanists (which will do the actual work).
+	botanist, err := botanistpkg.New(o)
+	if err != nil {
+		return formatError("Failed to create a Botanist", err)
+	}
+	seedCloudBotanist, err := cloudbotanistpkg.New(o, common.CloudPurposeSeed)
+	if err != nil {
+		return formatError("Failed to create a Seed CloudBotanist", err)
+	}
+
+	var (
+		defaultRetry = 30 * time.Second
+
+		f                     = flow.New("Backup Infrastructure creation").SetProgressReporter(o.ReportBackupInfrastructureProgress).SetLogger(o.Logger)
+		deployBackupNamespace = f.AddTask(botanist.DeployBackupNamespaceFromBackupInfrastructure, defaultRetry)
+		_                     = f.AddTask(seedCloudBotanist.DeployBackupInfrastructure, 0, deployBackupNamespace)
+	)
+	if e := f.Execute(); e != nil {
+		e.Description = fmt.Sprintf("Failed to reconcile Backup Infrastructure state: %s", e.Description)
+		return e
+	}
+
+	o.Logger.Infof("Successfully reconciled Backup Infrastructure state '%s'", o.BackupInfrastructure.Name)
+	return nil
+}
+
+// deleteBackupInfrastructure deletes a BackupInfrastructure entirely.
+func (c *defaultControl) deleteBackupInfrastructure(o *operation.Operation) *gardenv1beta1.LastError {
+
+	// We create botanists (which will do the actual work).
+	botanist, err := botanistpkg.New(o)
+	if err != nil {
+		return formatError("Failed to create a Botanist", err)
+	}
+
+	// We first check whether the namespace in the Seed cluster does exist - if it does not, then we assume that
+	// all resources have already been deleted. We can delete the BackupInfrastructure resource as a consequence.
+	namespace, err := botanist.K8sSeedClient.GetNamespace(common.GenerateBackupNamespaceName(o.BackupInfrastructure.Name))
+	if apierrors.IsNotFound(err) {
+		o.Logger.Infof("Did not find '%s' namespace in the Seed cluster - nothing to be done", common.GenerateBackupNamespaceName(o.BackupInfrastructure.Name))
+		return nil
+	}
+	if err != nil {
+		return formatError("Failed to retrieve the backup namespace in the Seed cluster", err)
+	}
+
+	seedCloudBotanist, err := cloudbotanistpkg.New(o, common.CloudPurposeSeed)
+	if err != nil {
+		return formatError("Failed to create a Seed CloudBotanist", err)
+	}
+
+	// We check whether the Backup namespace in the Seed cluster is already in a terminating state, i.e. whether
+	// we have tried to delete it in a previous run. In that case, we do not need to cleanup backup infrastructure resource because
+	// that would have already been done.
+	var (
+		cleanupBackupInfrastructureResources = namespace.Status.Phase != corev1.NamespaceTerminating
+		defaultRetry                         = 30 * time.Second
+
+		f                           = flow.New("Backup infrastructure deletion").SetProgressReporter(o.ReportBackupInfrastructureProgress).SetLogger(o.Logger)
+		destroyBackupInfrastructure = f.AddTaskConditional(seedCloudBotanist.DestroyBackupInfrastructure, 0, cleanupBackupInfrastructureResources)
```

Comment:
see above, why done by cloud botanist?

Reply:
Yes it is. It deploys terraform job with cloud specific configuration/charts.

Reply:
True, sorry.